### PR TITLE
Feature/api client dio

### PR DESCRIPTION
**Pull request by MayuRuru**

### CODE REVIEW

File: lib/src/core/config/config.dart
Comment (selankon) [inline lib/src/core/config/config.dart:35]: We are not using sentry! I guess we can delete this.

File: lib/src/core/helper/utils/constants.dart
Comment (selankon) [inline lib/src/core/helper/utils/constants.dart:39]: Why we need this??

File: lib/src/core/services/networking/api.dart
Comment (selankon) [inline lib/src/core/services/networking/api.dart]: Not sure, what is this??

File: lib/src/core/services/networking/api_endpoint.dart
Comment (selankon) [inline lib/src/core/services/networking/api_endpoint.dart:32]: We are not using all this paths

File: lib/src/core/services/networking/api_endpoint.dart
Comment (selankon) [inline lib/src/core/services/networking/api_endpoint.dart:41]: This is not own

File: lib/src/core/services/networking/api_endpoint.dart
Comment (selankon) [inline lib/src/core/services/networking/api_endpoint.dart:54]: Not own

File: lib/src/core/services/networking/api_endpoint.dart
Comment (selankon) [inline lib/src/core/services/networking/api_endpoint.dart:61]: Refactor from here to down to fetch our endpoints

File: lib/src/core/services/networking/api_interface.dart
Comment (selankon) [inline lib/src/core/services/networking/api_interface.dart:51]: Why this?

File: lib/src/core/services/networking/api_service_depecrated.dart
Comment (selankon) [inline lib/src/core/services/networking/api_service_depecrated.dart]: If this file is deprecated is better to delete it before merging to master

File: lib/src/core/services/networking/dio_service.dart
Comment (selankon) [inline lib/src/core/services/networking/dio_service.dart:1]: There are also a api_service_dio, which is the good one?

File: lib/src/core/services/networking/response_model.dart
Comment (selankon) [inline lib/src/core/services/networking/response_model.dart]: This is a good Idea

File: lib/src/core/services/networking/interceptors/refresh_token_interceptor.dart
Comment (selankon) [inline lib/src/core/services/networking/interceptors/refresh_token_interceptor.dart:3]: This is nice also

File: lib/src/core/services/networking/interceptors/api_interceptor.dart
Comment (selankon) [inline lib/src/core/services/networking/interceptors/api_interceptor.dart:1]: Big idea to work with interceptors, will help us a lot

**Review by selankon — CHANGES_REQUESTED** [review]
Please clean a little bit for a second review :)

File: lib/src/core/services/networking/api.dart
Comment (MayuRuru) [reply lib/src/core/services/networking/api.dart]: I thought it could be useful as we need to receive a token so users can register

File: lib/src/core/services/networking/api_endpoint.dart
Comment (MayuRuru) [reply lib/src/core/services/networking/api_endpoint.dart:32]: There is the option in EditProfile of updating password, but I get it is not this case. Shall I delete them or it is something it may be useful to implement?

File: lib/src/core/services/networking/api_endpoint.dart
Comment (MayuRuru) [reply lib/src/core/services/networking/api_endpoint.dart:41]: I copied the example as it looks neat to reply in our endpoints when they are defined

File: lib/src/core/services/networking/api_service_depecrated.dart
Comment (MayuRuru) [reply lib/src/core/services/networking/api_service_depecrated.dart]: I would like to double check which implementation for service are we using, this would be an alternative to dio_service

File: lib/src/core/services/networking/dio_service.dart
Comment (MayuRuru) [reply lib/src/core/services/networking/dio_service.dart:1]: api_service_dio implements api_interface, so methods fom dio_service are implemented there. But again, there are many ways of doing things and I am not sure which one is best

File: lib/src/core/services/networking/api_endpoint.dart
Comment (selankon) [reply lib/src/core/services/networking/api_endpoint.dart:32]: Better to delete whatever has no use yet

File: lib/src/core/services/networking/api_endpoint.dart
Comment (selankon) [reply lib/src/core/services/networking/api_endpoint.dart:41]: Same as above, if is not used, shouldn't be committed

**Comment by selankon** [comment]
I'm going to merge this and implement each feature on independent pr

**Pull request merged**
